{{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/consumers.py

```diff
@@ -1,27 +1,105 @@
-            await self.close()
+        await self.accept()
+
+    async def notify(self, event):
+        """
+        Handler for calls like::
+
+            channel_layer.group_send(group_name, {
+                'type': 'notify',  # This routes it to this handler.
+                'content': json_message,
+            })
+        """
+        if "content" in event:
+            await self.send_json(event["content"])
             return
+        # if "serializer" in event and "instance" in event and "inner_type" in event:
+        #     instance = self.get_instance(**event["instance"])
+        #     serializer = self.get_serializer(event["serializer"])
+        #     payload = {
+        #         "payload": serializer(
+        #             instance=instance, context=user_context(self.scope["user"])
+        #         ).data,
+        #         "type": event["inner_type"],
+        #     }
+        #     await self.send_json(payload)
+        #     return
 
-        await self.accept()
-        # Add this channel to the user-id group, so all browser windows
-        # where that user is logged in will get notifications:
-        user_id = self.scope["user"].id
-        await self.channel_layer.group_add(
-            f'user-{user_id}',
-            self.channel_name,
-        )
+    def get_instance(self, *, model, id):
+        Model = apps.get_model("api", model)
+        return Model.objects.get(pk=id)
+
+    # def get_serializer(self, serializer_path):
+    #     mod, serializer = serializer_path.rsplit(".", 1)
+    #     return getattr(import_module(mod), serializer)
 
-    async def disconnect(self, close_code):
-        user_id = self.scope["user"].id
-        await self.channel_layer.group_discard(
-            f'user-{user_id}',
-            self.channel_name,
+    async def receive_json(self, content, **kwargs):
+        # Just used to subscribe to notification channels.
+        is_valid = self.is_valid(content)
+        is_known_model = self.is_known_model(content.get("model", None))
+        has_good_permissions = self.has_good_permissions(content)
+        all_good = is_valid and is_known_model and has_good_permissions
+        if not all_good:
+            await self.send_json({"error": _("Invalid subscription.")})
+            return
+        group_name = CHANNELS_GROUP_NAME.format(
+            model=content["model"], id=content["id"]
+        )
+        self.groups.append(group_name)
+        await self.channel_layer.group_add(group_name, self.channel_name)
+        await self.send_json(
+            {
+                "ok": _("Subscribed to {model}.id = {id_}").format(
+                    model=content["model"], id_=content["id"]
+                )
+            }
         )
 
-    async def notify(self, event):
-        await self.send_json(event['content'])
+    def is_valid(self, content):
+        return content.keys() == {"model", "id"}
+
+    def is_known_model(self, model):
+        return model in KNOWN_MODELS
```

Comment:
I guess porting this over in a sort of template-y way makes sense. I intend to make this logic a reusable Django package in all my copious free time, but until I do that, this seems the best choice.